### PR TITLE
BAU: Add functionality to switch auth user roles in development via config var

### PR DESCRIPTION
**Pull request by harryyo**
### Add functionality to switch auth user roles in development, via DEBUG_USER_ROLE env var

- Refactor global auth functionality from create_app to new auth_protect() function
- Add DEBUG_USER_ROLE env var to development config (defaulting to "LEAD_ASSESSOR") 
- Add DEBUG_USER config var (in development config only) which creates a dict of user attributes with roles defined by DEBUG_USER_ROLE env var, to enable local development debugging of user roles

### CODE REVIEW

**Review by adamdavies1 — APPROVED** [review]
LGTM

File: config/envs/development.py [hunk 1]
```diff
@@ -11,21 +11,28 @@ class DevelopmentConfig(DefaultConfig):
...
+        "full_name": "Development User",
+        "email": "dev@example.com",
+        "roles": {
+            "LEAD_ASSESSOR": ["LEAD_ASSESSOR", "ASSESSOR", "COMMENTER"],
+            "ASSESSOR": ["ASSESSOR", "COMMENTER"],
+            "COMMENTER": ["COMMENTER"],
+        }.get(DEBUG_USER_ROLE),
+        "highest_role": DEBUG_USER_ROLE,
+    }
+
```
Comment (gio-karunakaran) [inline config/envs/development.py:35]: is this a universal test user for devs to use?

**Review by gio-karunakaran — APPROVED** [review]
LGTM 👍🏾

File: config/envs/development.py (hunk 1 above)
Comment (harryyo) [reply config/envs/development.py:35]: Yep that's the idea

**Pull request merged**